src/test/java/org/cactoos/list/ListEnvelopeTest.java

```diff
@@ -73,97 +61,113 @@ public void returnsListIteratorWithSupportedSet() {
+            "must be empty",
+            list,
+            new IsEqual<>(new ListOf<>("two", "one"))
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void setIsNotSupported() {
+    @Test
+    public void setIsDelegatedToTheOriginal() {
         final ListEnvelope<String> list = new StringList("one");
         list.set(0, "zero");
+        new Assertion<>(
+            "value must be changed",
+            list,
+            new HasValues<>("zero")
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void addIsNotSupported() {
+    @Test
+    public void addIsDelegated() {
         final ListEnvelope<String> list = new StringList("one");
         list.add("two");
+        new Assertion<>(
+            "value must be added",
+            list,
+            new HasValues<>("one", "two")
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void returnsSubListWithUnsupportedRemove() {
+    public void returnsSubListWithRemove() {
         final ListEnvelope<String> list = new StringList("one");
         list.subList(0, 1).remove(0);
+        new Assertion<>(
+            "must be empty after removal",
+            list,
+            new IsEmptyCollection<>()
+        ).affirm();
     }
 
-    @Test()
+    @Test
     public void returnsSubListWithSupportedSet() {
+        final List<String> list = new StringList("one");
+        list.subList(0, 1).set(0, "zero");
         new Assertion<>(
-            "subList.set() must throw exception",
-            () -> new StringList("one").subList(0, 1).set(0, "zero"),
-            new Throws<>(
-                new MatcherOf<>(
-                    (String msg) -> msg.equals("#set(): the list is read-only")
-                ),
-                UnsupportedOperationException.class
+            "subList.set() must change the original list",
+            list,
+            new HasValues<>(
+                "zero"
             )
         ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void returnsSubListWithUnsupportedAdd() {
+    @Test
+    public void addsAtGivenIndex() {
         final ListEnvelope<String> list = new StringList("one");
-        list.subList(0, 1).add("two");
+        list.add(0, "two");
+        new Assertion<>(
+            "must add value",
+            list,
+            new HasValues<>("two")
+        ).affirm();
+    }
+
+    @Test
+    public void getsAtGivenIndex() {
+        final ListEnvelope<String> list = new StringList("one");
+        new Assertion<>(
+            "must add value",
+            list.get(0),
+            new IsEqual<>("one")
+        ).affirm();
+    }
+
+    @Test
+    public void getsLastIndexOfValue() {
+        final ListEnvelope<String> list = new StringList("one");
+        list.add(1, "one");
+        new Assertion<>(
+            "must be last index",
```

Comment:
@andreoss Please, add more detail in this assertion message (as explained before).